test/promql/tests/data_plane/gitops_sync_test.yaml

```diff
@@ -0,0 +1,84 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.gitops_sync_alerts.yaml
+
+tests:
+  # Alerted cases:
+  - interval: 1m
+    input_series:
+
+      # GitOps service is experiencing application sync failure rate of less than 95%, so alert
+      - series: 'argocd_app_info{sync_status="Synced", namespace="gitops-service-argocd", pod="pod-1", source_cluster="cluster01"}'
+        values: '1+1x60'
+      - series: 'argocd_app_info{sync_status!="", namespace="gitops-service-argocd", pod="pod-1", source_cluster="cluster01"}'
+        values: '20+20x60'
+
+    alert_rule_test:
+      - eval_time: 60m
+        alertname: GitOpsApplicationSyncErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: gitops-service-argocd
+              pod: pod-1
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                The synced status percentage of all GitOps service applications is <95%
+              description: >-
+                The percentage total of all the Argo CD applications on cluster cluster01 having a sync
+                status of Synced is less than 95%.
+              runbook_url: https://gitlab.cee.redhat.com/rhtap/docs/sop/-/blob/main/gitops/deploy-from-git-to-k8s.md
+
+  - interval: 1m
+    input_series:
+
+      # GitOps Service experienced less than 95% synced rate in applications, should alert
+      - series: 'argocd_app_info{sync_status="Synced", namespace="gitops-service-argocd", pod="pod-1", source_cluster="cluster01"}'
+        values: '1+1x60'
+      - series: 'argocd_app_info{sync_status!="", namespace="gitops-service-argocd", pod="pod-1", source_cluster="cluster01"}'
+        values: '1+1x50 20+1x10'
+
+    alert_rule_test:
+      - eval_time: 65m
```

Comment:
We have 62m-worth of samples in this test (1 + 50 + 1 + 10), but we take the measurement at 65m. Unless we want to test the condition in which we don't have samples, we should have the evaluation happen within a time frame for which we have samples.

Also, I think we can do with 6m-worth of samples instead of ~1h. It will make things easier to comprehend.

Reply:
Ah, it should have been 60, like the first test. It was a copy and paste from before the change.  I can reduce it down to 6m

Reply:
I committed a change for this and used the expanded notation instead. Also, the sequences are based on the sum and my interpretation of it, which from the other comment, might need to be changed.